tenets/codelingo/go/init/codelingo.yaml

```diff
@@ -0,0 +1,19 @@
+tenets:
+  - name: init
+    doc: |
+      Check that no inits functions are present in Go code.
+    flows:
+      codelingo/review:
+        comment: Init functions cause an import to have a side effects, and side effects are hard to test, reduce readability and increase the complexity of code.
+    query: |
+      import codelingo/ast/go
+      
+      go.file(depth = any):
+        go.decls:
+          go.func_decl:
+            @ review.comment
+            go.ident:
+              name == "init"
+            go.func_type:
```

Comment:
This is an indirect way of finding the init func. You'll still hit a false positive for

```
func (r *typ) init() {
// ...
}
```

i.e. remove the field_list condition. Can we distinguish between methods and functions? Functions should not have a receiver.

Reply:
Yup I used that to check the arity of the function was == 0. Not positive if this is also the way to check for receivers